File: src/types/note.ts
export type NoteColor = 'yellow' | 'green' | 'blue' | 'pink' | 'purple' | 'orange';

export interface StickyNote {
  id: number;
  title: string;
  content: string;
  color: NoteColor;
  pinned: boolean;
  created_at: string;
  updated_at: string;
  file_path?: string | null;
}

export interface CreateNote {
  title?: string;
  content?: string;
  color?: NoteColor;
}

export interface UpdateNote {
  id: number;
  title?: string;
  content?: string;
  color?: NoteColor;
}

export interface ExportNotesResult {
  folder: string;
  files: string[];
}

export interface NoteTemplate {
  id: string;
  name: string;
  title: string;
  body: string;
  source: 'builtin' | 'file';
  path?: string | null;
}

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StickyNote {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub color: String,
    pub pinned: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateNote {
    pub title: Option<String>,
    pub content: Option<String>,
    pub color: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateNote {
    pub id: i64,
    pub title: Option<String>,
    pub content: Option<String>,
    pub color: Option<String>,
}
